components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt

```diff
@@ -37,34 +38,53 @@ class WebExtensionBrowserMenuBuilder(
     private val store: BrowserStore,
     private val style: Style = Style(),
     private val onAddonsManagerTapped: () -> Unit = {},
-    private val appendExtensionSubMenuAtStart: Boolean = false
+    private val appendExtensionSubMenuAtStart: Boolean = false,
+    private val showAddonsInMenu: Boolean = true
 ) : BrowserMenuBuilder(items, extras, endOfMenuAlwaysVisible) {
 
+    private val finalList = items.toMutableList()
+
     /**
      * Builds and returns a browser menu with combination of [items] and web extension browser actions.
      */
     override fun build(context: Context): BrowserMenu {
         val extensionMenuItems =
             WebExtensionBrowserMenu.getOrUpdateWebExtensionMenuItems(store.state, store.state.selectedTab)
 
-        val finalList = items.toMutableList()
-
         val filteredExtensionMenuItems = extensionMenuItems.filter { webExtensionBrowserMenuItem ->
-            // Check if we have a placeholder
-            val index = finalList.indexOfFirst { browserMenuItem ->
-                (browserMenuItem as? WebExtensionPlaceholderMenuItem)?.id == webExtensionBrowserMenuItem.id
-            }
-            // Replace placeholder with corresponding web extension, and remove it from extensions menu list
-            if (index != -1) {
-                webExtensionBrowserMenuItem.setIconTint(
-                    (finalList[index] as? WebExtensionPlaceholderMenuItem)?.iconTintColorResource
-                )
-                finalList[index] = webExtensionBrowserMenuItem
-            }
-            index == -1
+            replaceMenuPlaceholderWithExtensions(webExtensionBrowserMenuItem)
         }
 
-        val webExtMenuItem = if (filteredExtensionMenuItems.isNotEmpty()) {
+        val menuItems = if (showAddonsInMenu) {
+            showAddonsMenuItems(context, filteredExtensionMenuItems)
+        } else {
+            finalList
+        }
+
+        val adapter = BrowserMenuAdapter(context, menuItems)
+        return BrowserMenu(adapter)
+    }
+
+    private fun replaceMenuPlaceholderWithExtensions(menuItem: WebExtensionBrowserMenuItem): Boolean {
+        // Check if we have a placeholder
+        val index = items.indexOfFirst { browserMenuItem ->
+            (browserMenuItem as? WebExtensionPlaceholderMenuItem)?.id == menuItem.id
+        }
+        // Replace placeholder with corresponding web extension, and remove it from extensions menu list
+        if (index != -1) {
+            menuItem.setIconTint(
+                (finalList[index] as? WebExtensionPlaceholderMenuItem)?.iconTintColorResource
+            )
+            finalList[index] = menuItem
+        }
+        return index == -1
+    }
+
+    private fun showAddonsMenuItems(
```

Comment:
nit: maybe `createAddonsMenuItems`, as we are creating a new list, what do you think? :)